method.rb

```diff
@@ -0,0 +1,113 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i]
+      i += 1
+    end
+    output_array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i], i
+      i += 1
+    end
+    output_array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    output_array = []
+    my_each do |item|
+      output_array << item if (yield item) == true
+    end
+    output_array
+  end
+
+  def my_all?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity
+    status_cumulator = true
```

Comment:
* [ ] There should be a working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
 #### compare the following examples by replacing the custom method with the original method in each example and rectify the error that arises.
``` p [1, 2i, 3.14].my_all?(Numeric) #=> must return true```

``` p [nil, true, 99].my_all? #=>must return false```

``` p [].my_all? #=> true```